src/internal/Util.ts

```diff
@@ -19,4 +29,47 @@ export class Util {
     }
     return undefined;
   }
+
+  /**
+   * Whether or not the passed object is already a date. (Either a [[Moment]] object, or
+   * a JavaScript [[Date]] object.)
+   */
+  public static isDateObject(date: any) {
+    return moment.isMoment(date) || date instanceof Date;
+  }
+
+  /**
+   * Create a [[Moment]] from any form of date (JavaScript [[Date]], [[Moment]], or epoch).
+   * [[Moment]] dates in OpenNMS.js will always be converted internally to UTC to avoid time
+   * zone issues.
+   */
+  public static toDate(date: Date|Moment|string|number): Moment {
+    if (date === undefined || date === null) {
+      return undefined;
+    } else if (moment.isMoment(date)) {
+      return (date as Moment).utc();
+    } else if (typeof(date) === 'number' || date instanceof Date
+      || typeof(date) === 'string' || date instanceof String) {
+      return moment(date).utc();
+    } else {
+      throw new OnmsError('Unable to parse type "' + typeof(date) + '" as a date.');
+    }
+  }
+
+  /**
+   * Create a date string from any form of date (JavaScript [[Date]], [[Moment]], or epoch).
+   * Date strings in OpenNMS.js will always be converted internally to UTC to avoid time
+   * zone issues.
+   */
+  public static toDateString(date: Date|Moment|number) {
+    if (date === undefined || date === null) {
```

Comment:
I suppose so.  Lemme refactor a little.  It still does have a slightly different "requirement" than `toDate()` (which I want to support strings), because I don't want `toDateString()` to coerce a string into a date without the API-user's explicit intent.